### PR TITLE
Deepspeech CuDNNLSTM layer fix

### PR DESCRIPTION
**Pull request by sourabh2k15**
tensorboard showing deepspeech with new layer hits 11.5% valid/WER in 56.6k steps taking ~14hours : https://tensorboard.dev/experiment/4A18D2BtQI6A4ENzr0seSg/#scalars&_smoothingWeight=0 

target was 11.62% valid/WER and timing is on par with torch deepspeech runs.

### CODE REVIEW

**Comment by sourabh2k15** [comment]
command to run workload : 

```
python3 submission_runner.py     --framework=jax     --workload=librispeech_deepspeech --submission_path=reference_algorithms/target_setting_algorithms/jax_nadamw.py     --tuning_search_space=reference_algorithms/target_setting_algorithms/librispeech_deepspeech/tuning_search_space.json --data_dir=../data/data/ --librispeech_tokenizer_vocab_path=../data/spm_model.vocab --num_tuning_trials=1 --experiment_dir=../experiment_runs/  --experiment_name=librispeech_deepspeech_new_layer_6_lstms_memory_fix4 --max_global_steps=10

```

File: submission_runner.py [hunk 1]
```diff
@@ -23,28 +23,23 @@
...
-from algorithmic_efficiency.pytorch_utils import pytorch_setup
-from algorithmic_efficiency.pytorch_utils import sync_ddp_time
+from algorithmic_efficiency.profiler import PassThroughProfiler, Profiler
+from algorithmic_efficiency.pytorch_utils import (pytorch_init, pytorch_setup,
+                                                  sync_ddp_time)
 
 # Hide any GPUs form TensorFlow. Otherwise TF might reserve memory and make
 # it unavailable to JAX.
 tf.config.set_visible_devices([], 'GPU')
+os.environ["XLA_FLAGS"] = "--xla_gpu_enable_triton_gemm=false"
```
Comment (znado) [inline submission_runner.py:42]: can we add this only for deepspeech?

**Review by znado — APPROVED** [review]

File: submission_runner.py (hunk 1 above)
Comment (sourabh2k15) [reply submission_runner.py:44]: resolved offline, added a comment to disable only for deepspeech if it works fine for other workloads

**Comment by georgedahl** [comment]
@sourabh2k15 can you run Priya's test script(s) and fix any issues that arise on the other workloads due to the jax, jaxlib, and cudnn upgrades before merging?

**Comment by priyakasimbeg** [comment]
@sourabh2k15 could you make sure that the upgrades in this PR work with the other workloads by running all jax workloads for at least one training algorithm for ~10% of the total steps. 
You can use run the following standalone script on one of your VMs in a tmux session https://github.com/priyakasimbeg/algo-efficiency-timing/blob/main/run_all_workloads.py. 
Example command:
```
python run_all_workloads.py --framework jax \
--algorithm adamw \
--experiment_basename <experiment_basename e.g. jax_upgrades> \
--docker_image_url <url_for_docker_image> \
--tag <some_docker_tag> \
--run_percentage 10 
```
The results along with the logs will be saved to gs://mlcommons-runs/<experiment_basename>. I think at the very least we should record here what breaks before submitting the PR.

**Comment by sourabh2k15** [comment]
I wasn't able to make `run_all_workloads.py` work so I manually ran some workloads using adam for 3k steps at current PR, uploaded results in mlcommons-runs bucket 

deepspeech and ogbg completed successfully , others are running

**Comment by sourabh2k15** [comment]
uploaded fastmri, imagenet_resnet and imagenet_vit as well under jax_upgrade/ in mlcommons-runs bucket 

all of those ran without issues

**Comment by sourabh2k15** [comment]
forgot criteo, testing that as soon as I get data on VM ...

**Comment by sourabh2k15** [comment]
criteo also works fine, please approve this PR

**Review by znado — APPROVED** [review]

**Pull request merged**